### PR TITLE
Medical Damage - Add setting for unconscious chance from pain

### PR DESCRIPTION
**Pull request by TyroneMF**
**When merged this pull request will:**
- Changes the default 10% chance of unconsciousness from pain to a slider.

### CODE REVIEW

File: addons/medical_damage/initSettings.sqf [hunk 1]
```diff
@@ -24,3 +24,12 @@
     [0, 25, 1, 2],
     true
 ] call CBA_settings_fnc_init;
+
+[
+    QEGVAR(medical,painUnconsciousChance),
+    "SLIDER",
+    [LSTRING(painUnconsciousChance_DisplayName), LSTRING(painUnconsciousChance_Description)],
+    ELSTRING(medical,Category),
+    [0, 1, 0.1, 2],
```
Comment (mharis001) [inline addons/medical_damage/initSettings.sqf:33]: Use a percentage slider; 5th argument set as `true`.

File: addons/medical_damage/stringtable.xml [hunk 2]
```diff
@@ -687,5 +687,11 @@
             <Polish>Ustawia próg obrażeń jakie może otrzymać jednostka przed utratą przytomności. (0 jako ustawienie domyślne misji)</Polish>
             <Turkish>Bilinçsiz duruma geçmeden önce bir birimin alabileceği hasar miktarını ayarlar. (Görev varsayılanı için 0)</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_painUnconsciousChance_DisplayName">
```
Comment (mharis001) [inline addons/medical_damage/stringtable.xml:690]: Capitalize first letter to match other stringtable keys here and for description: `STR_ACE_Medical_Damage_PainUnconsciousChance_DisplayName`.

File: addons/medical_damage/stringtable.xml [hunk 3]
```diff
@@ -687,5 +687,11 @@
...
+            <English>Pain Unconscious Chance</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_painUnconsciousChance_Description">
+            <English>Sets the chance of falling unconscious if pain is above the threshold.</English>
```
Comment (mharis001) [inline addons/medical_damage/stringtable.xml:694]: For other "chance settings" we use "The probability..." and I think for consistency we should have the same here. Something like: "The probability of a person falling unconscious when their pain is above the tolerance threshold."

File: addons/medical_damage/initSettings.sqf [hunk 4]
```diff
@@ -24,3 +24,12 @@
     [0, 25, 1, 2],
     true
 ] call CBA_settings_fnc_init;
+
+[
+    QEGVAR(medical,painUnconsciousChance),
+    "SLIDER",
+    [LSTRING(painUnconsciousChance_DisplayName), LSTRING(painUnconsciousChance_Description)],
```
Comment (mharis001) [inline addons/medical_damage/initSettings.sqf:31]: Capitalize first letter here as well.

**Review by mharis001 — APPROVED** [review]

File: addons/medical_damage/initSettings.sqf [hunk 5]
```diff
@@ -24,3 +24,12 @@
...
 ] call CBA_settings_fnc_init;
+
+[
+    QEGVAR(medical,painUnconsciousChance),
+    "SLIDER",
+    [LSTRING(PainUnconsciousChance_DisplayName), LSTRING(PainUnconsciousChance_Description)],
+    ELSTRING(medical,Category),
+    [0, 1, 0.1, 2, true],
+    true
+] call CBA_settings_fnc_init;
```
Comment (Dystopian) [inline addons/medical_damage/initSettings.sqf:35]: `CBA_settings_fnc_init` should be replaced with `CBA_fnc_addSetting` here and above

Comment (jonpas) [reply addons/medical_damage/initSettings.sqf:35]: I'd say do that in separate PR ACE-wide. All medical still uses the old function.

File: addons/medical_damage/stringtable.xml [hunk 6]
```diff
@@ -687,5 +687,11 @@
             <Polish>Ustawia próg obrażeń jakie może otrzymać jednostka przed utratą przytomności. (0 jako ustawienie domyślne misji)</Polish>
             <Turkish>Bilinçsiz duruma geçmeden önce bir birimin alabileceği hasar miktarını ayarlar. (Görev varsayılanı için 0)</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_DisplayName">
+            <English>Pain Unconscious Chance</English>
```
Comment (veteran29) [inline addons/medical_damage/stringtable.xml:691]:
```suggestion
            <English>Pain Unconscious Chance</English>
            <Polish>Szansa na nieprzytomność przez ból</Polish>
```

File: addons/medical_damage/stringtable.xml [hunk 7]
```diff
@@ -687,5 +687,11 @@
...
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_Description">
+            <English>The probability of a person falling unconscious when their pain is above the tolerance threshold.</English>
```
Comment (veteran29) [inline addons/medical_damage/stringtable.xml:694]:
```suggestion
            <English>The probability of a person falling unconscious when their pain is above the tolerance threshold.</English>
            <Polish>Szansa że osoba straci przytomność gdy jej ból jest powyżej tolerowalnego progu.</Polish>
```

**Review by veteran29 — APPROVED** [review]

File: addons/medical_damage/stringtable.xml [hunk 8]
```diff
@@ -687,5 +687,13 @@
             <Polish>Ustawia próg obrażeń jakie może otrzymać jednostka przed utratą przytomności. (0 jako ustawienie domyślne misji)</Polish>
             <Turkish>Bilinçsiz duruma geçmeden önce bir birimin alabileceği hasar miktarını ayarlar. (Görev varsayılanı için 0)</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_DisplayName">
+            <English>Pain Unconscious Chance</English>
+            <Polish>Szansa na nieprzytomność przez ból</Polish>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_Description">
+            <English>The probability of a person falling unconscious when their pain is above the tolerance threshold.</English>
```
Comment (Bummeri) [inline addons/medical_damage/stringtable.xml:695]: This is not completely accurate. The chance is not to be knocked out when you are just over the pain threshold. It is the chance to be knocked when you receive damage and if then you go over the tolerance threshold. It is checked only then. So accurate description would be:

"Chance of being knocked unconscious when receiving damage and exceeding pain threshold"

Comment (jonpas) [reply addons/medical_damage/stringtable.xml:695]: Just add "when receiving damage". If we want to change "probability" to "chance", that should be done in separate PR for all settings. But I don't see why.

File: addons/medical_damage/stringtable.xml [hunk 9]
```diff
@@ -687,5 +687,13 @@
             <Polish>Ustawia próg obrażeń jakie może otrzymać jednostka przed utratą przytomności. (0 jako ustawienie domyślne misji)</Polish>
             <Turkish>Bilinçsiz duruma geçmeden önce bir birimin alabileceği hasar miktarını ayarlar. (Görev varsayılanı için 0)</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_DisplayName">
+            <English>Pain Unconscious Chance</English>
+            <Polish>Szansa na nieprzytomność przez ból</Polish>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_PainUnconsciousChance_Description">
+            <English>The probability of a person falling unconscious when their pain is above the tolerance threshold upon receiving damage.</English>
+            <Polish>Szansa że osoba straci przytomność gdy jej ból jest powyżej tolerowalnego progu.</Polish>
```
Comment (veteran29) [inline addons/medical_damage/stringtable.xml:696]: PL translation update to match EN.
```suggestion
            <Polish>Szansa że osoba straci przytomność gdy jej ból jest powyżej tolerowalnego progu podczas otrzymywania obrażeń.</Polish>
```

**Pull request merged**